### PR TITLE
Media Library: Improve loading state

### PR DESCRIPTION
**Pull request by frosty**
This PR improves the appearance of the media library when a user first visits their Media Library. Currently, they'll see the no results view saying "You have no media" even if they do have some. This PR adds a new 'fetching media...' animated no results view to keep the user better informed about what's happening when we're fetching media the first time.

I had to add two new delegate methods to `MediaPicker-iOS` to let me update the state correctly. You can find that here: https://github.com/wordpress-mobile/MediaPicker-iOS/pull/179

This PR currently points the Podfile at that branch. Once that's been merged and released, I'll update the pod file here before merging. I've set the 'do not merge' label as a reminder.

To test:

* Build and run
* Visit the Media Library section for a site with no media, or a site you haven't yet viewed in the app
* Check that when the media section first loads, you see 'Fetching media..." displayed
* Check that when you upload the first media item to an empty library, the no results view is removed from view.

Needs review: @kurzee

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift [hunk 1]
```diff
@@ -102,7 +103,11 @@ class MediaLibraryViewController: UIViewController {
 
         registerChangeObserver()
 
-        updateViewState(for: pickerDataSource.totalAssetCount)
+        if isLoading {
+            updateNoResultsForFetching()
+        } else {
+            updateViewState(for: pickerDataSource.totalAssetCount)
```
Comment (kurzee) [inline WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift]: Hmm, should we combine these two methods some how? Perhaps `updateNoResultsForFetching()` is only ever called, if needed, within the call to `updateViewState(for:)`.

**Comment by frosty** [comment]
@kurzee Thanks! I've just incorporated the fetching no results call into the main one – see what you think.

**Comment by kurzee** [comment]
>I've just incorporated the fetching no results call into the main one – see what you think.

Looks good!

There is one other small issue, in general:

1. Uninstall the app to wipe out all cached media.
2. Log in.
3. Open a site that has a large amount of Media to fetch.
4. As soon as it begins fetching, switch tabs.
5. Switch back to tab with the Media fetching.
6. Notice the activity spinner has stopped spinning while it continues loading.

**Comment by frosty** [comment]
@kurzee I've updated the MediaPicker pod. As discussed, the other issue you found seems to affect `develop` too, so I can take a look at resolving it elsewhere.

**Pull request merged**